### PR TITLE
Code beautification in bootloader.py

### PR DESCRIPTION
**Pull request by cbosdo**
Addressing @aplanas comments in PR #35

### CODE REVIEW

File: virtualization-host-formula/virtualization-host/_states/bootloader.py [hunk 1]
```diff
@@ -39,15 +39,14 @@ def grub_set_default(name):
...
-                ret['result'] = False
-            ret['result'] = True
-        else:
+        if __opts__['test']:
             ret['result'] = None
+            return ret
+        try:
+            out = __salt__['cmd.run']('grub2-mkconfig -o /boot/grub2/grub.cfg', raise_err=True)
+        except CommandExecutionError:
+            ret['comment'] = 'Failed to run grub2-mkconfig: {0}'.format(out)
```
Comment (aplanas) [inline virtualization-host-formula/virtualization-host/_states/bootloader.py]: Right, do you miss a `return` here?

File: virtualization-host-formula/virtualization-host/_states/bootloader.py [hunk 2]
```diff
@@ -39,15 +39,14 @@ def grub_set_default(name):
...
+        ret['result'] = True
     else:
```
Comment (aplanas) [inline virtualization-host-formula/virtualization-host/_states/bootloader.py]: IIUC this `else` comes from `if len(filtered_entries) > 0`, you can avoid the indentation using the reverse check and adding the error message and the `return` in the `if` leg, avoiding the `else`

The pattern is like this:

```
if something is wrong1:
  ret['comment'] = 'wrong1'
  return ret

if something is wrong2:
  ...
  return ret

if test:
  ret['result'] = 'some changes are needed'
  ret['changes'] = ['maybe my list of possible changes']

do_changes()

if something fails:
  ..
  return ret

ret['changes'] = [...]
ret['result'] = True
return ret
```

File: virtualization-host-formula/virtualization-host/_states/bootloader.py [hunk 3]
```diff
@@ -30,26 +30,27 @@ def grub_set_default(name):
...
+    # Regenerate grub config
+    out = None
+    if __opts__['test']:
+        ret['result'] = None
+        return ret
+    try:
+        out = __salt__['cmd.run']('grub2-mkconfig -o /boot/grub2/grub.cfg', raise_err=True)
+    except CommandExecutionError:
+        ret['comment'] = 'Failed to run grub2-mkconfig: {0}'.format(out)
+        return ret
```
Comment (aplanas) [inline virtualization-host-formula/virtualization-host/_states/bootloader.py]: I think that as it is, is perfect. :+1: 

For completeness, if you really hate to have two `return` so close, you can add an `else` clause where you set the `result` to `True`, and reuse the last return.

Comment (cbosdo) [reply virtualization-host-formula/virtualization-host/_states/bootloader.py]: bah, that will do like this... the point is not to create a master art piece, just to get it readable enough.

Comment (aplanas) [reply virtualization-host-formula/virtualization-host/_states/bootloader.py]: I agree.

**Pull request merged**